src/sc/sidechaintypes.cpp

```diff
@@ -1,139 +1,329 @@
+#ifdef BITCOIN_TX
+    CFieldElement::CFieldElement() {};
+    void CFieldElement::SetByteArray(const std::vector<unsigned char>& byteArrayIn) {};
+    void CFieldElement::SetNull() {};
+    bool CFieldElement::IsNull() const {return false;};
+    field_t* CFieldElement::GetFieldElement() const {return nullptr;};
+    bool CFieldElement::IsValid() const {return false;};
+    std::string CFieldElement::GetHexRepr() const {return std::string{};};
+    CFieldElement CFieldElement::ComputeHash(const CFieldElement& lhs, const CFieldElement& rhs) { return CFieldElement{}; }
+    const std::vector<unsigned char>&  CFieldElement::GetByteArray() const { static std::vector<unsigned char> dummy{}; return dummy;}
+#else
+CFieldElement::CFieldElement(): byteVector() { SetNull(); }
+
+CFieldElement::CFieldElement(const std::vector<unsigned char>& byteArrayIn) : byteVector()
 {
-    //TENTATIVE IMPLEMENTATION, BEFORE ACTUAL ONE
-    return (nBits > 0);
+    this->SetByteArray(byteArrayIn);
 }
 
-CompressedFieldElementConfig::CompressedFieldElementConfig(int32_t nBitsIn): CustomFieldConfig(), nBits(nBitsIn)
+void CFieldElement::SetByteArray(const std::vector<unsigned char>& byteArrayIn)
 {
-    if (!isBitsLenghtValid())
-        throw std::invalid_argument("CompressedFieldElementConfig size must be strictly positive");
+    assert(byteArrayIn.size() == CFieldElement::ByteSize());
+    this->byteVector = byteArrayIn;
 }
 
-int32_t CompressedFieldElementConfig::getBitSize() const
+CFieldElement::CFieldElement(const uint256& value)
 {
-    //TENTATIVE IMPLEMENTATION, BEFORE ACTUAL ONE
-    return nBits;
+    std::copy(value.begin(), value.end(), this->byteVector.begin());
 }
 
-//----------------------------------------------------------------------------------
-bool CompressedMerkleTreeConfig::isTreeHeightValid() {
-    //TENTATIVE IMPLEMENTATION, BEFORE ACTUAL ONE
-    return ((treeHeight >= 0) && (treeHeight < MAX_TREE_HEIGHT));
+void CFieldElement::SetNull()
+{
+    byteVector.resize(0);
 }
 
-CompressedMerkleTreeConfig::CompressedMerkleTreeConfig(int32_t treeHeightIn): CustomFieldConfig(), treeHeight(treeHeightIn)
+bool CFieldElement::IsNull() const { return byteVector.empty();}
+
+const std::vector<unsigned char>&  CFieldElement::GetByteArray() const
 {
-	if (!isTreeHeightValid())
-        throw std::invalid_argument(
-            std::string("CompressedMerkleTreeConfig height=" + std::to_string(treeHeight) +
-            ", must be in the range [0, ") + std::to_string(MAX_TREE_HEIGHT) + std::string(")"));
+    return byteVector;
 }
 
-int32_t CompressedMerkleTreeConfig::getBitSize() const
+field_t* CFieldElement::GetFieldElement() const
 {
-    //TENTATIVE IMPLEMENTATION, BEFORE ACTUAL ONE
-    int nBytes = (treeHeight == -1)? 0: 1 << treeHeight;
-    return nBytes * CHAR_BIT;
+    return zendoo_deserialize_field(&this->byteVector[0]);
 }
 
+uint256 CFieldElement::GetLegacyHashTO_BE_REMOVED() const
```

Comment:
seems, not used

Reply:
Keep it as placeholder for discussion about scTxCommitmentTreeHash type